### PR TITLE
Checkov api lambda iam issues

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/278

- moved s3 action to its own statement and restricted to the csv upload bucket
- deleted PinPoint and SES actions
- had issues with the plan not picking out the new `common` outputs, so upgraded terragrunt and used the `mock_outputs_merge_with_state` attribute (as discussed below). Unclear if we need to do this in prod.

# Test instructions | Instructions pour tester la modification

# Help requested | Aide requise

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

Does api need `sns:Publish` and `securityhub:BatchImportFindings` ?

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: .github/workflows/terragrunt_plan_staging.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
   AWS_SECRET_ACCESS_KEY: ${{ secrets.STAGING_AWS_SECRET_ACCESS_KEY }}
   AWS_REGION: ca-central-1
   TERRAFORM_VERSION: 0.14.4
-  TERRAGRUNT_VERSION: 0.31.0
+  TERRAGRUNT_VERSION: 0.35.13
```
Comment (sastels) [inline .github/workflows/terragrunt_plan_staging.yml:17]: needed for `mock_outputs_merge_with_state` attribute

File: .github/workflows/terragrunt_plan_staging.yml [hunk 2]
```diff
@@ -72,7 +72,7 @@ jobs:
           filters: |
             common:
               - '.github/workflows/terragrunt-plan-staging.yml'
-              - 'env/common/**'
+              - 'aws/common/**'
```
Comment (sastels) [inline .github/workflows/terragrunt_plan_staging.yml:75]: typo

File: aws/lambda-api/iam.tf [hunk 3]
```diff
@@ -47,14 +47,18 @@ data "aws_iam_policy_document" "api_policies" {
   statement {
     effect = "Allow"
     actions = [
-      "mobiletargeting:*",
-      "ses:SendEmail",
-      "ses:SendRawEmail",
```
Comment (sastels) [inline aws/lambda-api/iam.tf:52]: Deleted these 3 actions (in k8s api and celery have the same iam policy, this was just copied from that). Note that the `mobiletargeting` one is for PinPoint.

File: .github/workflows/terragrunt_plan_staging.yml (hunk 2 above)
Comment (patheard) [reply .github/workflows/terragrunt_plan_staging.yml:75]: good catch on the missing `aws/common/**`, but I think you'll also want `env/common/**` to catch changes to the `terragrunt.hcl` file in there.

**Review by patheard — COMMENTED** [review]
Looks good Steve!  I just had a question about keeping `env/common/**` in the path filter checking for changes.

I was also curious about this in the plan:

```hcl
  # module.notify_slack_critical.module.lambda.null_resource.archive[0] must be replaced
-/+ resource "null_resource" "archive" {
      ~ id       = "8695568181817461827" -> (known after apply)
      ~ triggers = { # forces replacement
          ~ "timestamp" = "1639579846555974000" -> "<WARNING: Missing lambda zip artifacts wouldn't be restored>"
            # (1 unchanged element hidden)
        }
    }
```

It seems to be related to the `terraform-aws-modules/notify-slack/aws` module using an old version of the `terraform-aws-modules/lambda/aws` module:

terraform-aws-modules/terraform-aws-lambda#179

File: .github/workflows/terragrunt_plan_staging.yml (hunk 2 above)
Comment (sastels) [reply .github/workflows/terragrunt_plan_staging.yml:75]: ah, yeah, there's no `env/common` actually, but probably the typo was actually supposed to be `env/staging/common/**` 👍

**Review by patheard — APPROVED** [review]

**Comment by sastels** [comment]
> It seems to be related to the terraform-aws-modules/notify-slack/aws module using an old version of the terraform-aws-modules/lambda/aws module:

I don't know if this is true - we're using the latest version of `notify-slack` which uses the version of `lambda` from Dec 9
https://github.com/cds-snc/notification-terraform/runs/4538085828?check_suite_focus=true#step:6:50

**Comment by patheard** [comment]
> > It seems to be related to the terraform-aws-modules/notify-slack/aws module using an old version of the terraform-aws-modules/lambda/aws module:
> 
> I don't know if this is true - we're using the latest version of `notify-slack` which uses the version of `lambda` from Dec 9 https://github.com/cds-snc/notification-terraform/runs/4538085828?check_suite_focus=true#step:6:50

Ah, you're totally right.  When I looked at the module source, I had misread this line in the source as an older version:
https://github.com/terraform-aws-modules/terraform-aws-notify-slack/blob/master/main.tf#L73

**Pull request merged**